GVFS/GVFS.Installer.Mac/CreateMacInstaller.sh

```diff
@@ -108,13 +112,34 @@ function CopyBinariesToInstall()
     cd $currentDirectory
 }
 
-function CreateInstaller()
+function CreateVFSForGitInstaller()
 {
-    pkgBuildCommand="/usr/bin/pkgbuild --identifier $INSTALLERPACKAGEID --scripts \"${SCRIPTSPATH}\" --root \"${STAGINGDIR}\" \"${BUILDOUTPUTDIR}\"$INSTALLERPACKAGENAME.pkg"
+    pkgBuildCommand="/usr/bin/pkgbuild --identifier $INSTALLERPACKAGEID --scripts \"${SCRIPTSPATH}\" --root \"${STAGINGDIR}\" \"${PACKAGESTAGINGDIR}/$INSTALLERPACKAGENAME.pkg\""
     eval $pkgBuildCommand || exit 1
 }
 
-function CreateMetaInstaller()
+function CreateVFSForGitDistribution()
+{
+	# 1.  Update Distribution.xml
+	# 1.1 Substitute Git placeholders with actual values
+    /usr/bin/sed -e "s|GIT_CHOICE_OUTLINE_PLACEHOLDER||g" "$SCRIPTSPATH/Distribution.xml" > "${BUILDOUTPUTDIR}/Distribution.updated.xml"
+	/usr/bin/sed -i.bak "s|GIT_CHOICE_ID_PLACEHOLDER||g" "${BUILDOUTPUTDIR}/Distribution.updated.xml"
+	/usr/bin/sed -i.bak "s|GIT_PKG_REF_PLACEHOLDER||g" "${BUILDOUTPUTDIR}/Distribution.updated.xml"
+	
+	# 1.2 Substitute VFSForGit placeholders with actual values
+	/usr/bin/sed -i.bak "s|VFSFORGIT_VERSION_PLACHOLDER|$PACKAGEVERSION|g" "${BUILDOUTPUTDIR}/Distribution.updated.xml"
+	/usr/bin/sed -i.bak "s|VFSFORGIT_PKG_NAME_PLACEHOLDER|$INSTALLERPACKAGENAME.pkg|g" "${BUILDOUTPUTDIR}/Distribution.updated.xml"
+	
+	/bin/rm -f "${BUILDOUTPUTDIR}/Distribution.updated.xml.bak"
+	
+    buildVFSForGitDistCmd="/usr/bin/productbuild --distribution \"${BUILDOUTPUTDIR}/Distribution.updated.xml\" --package-path \"$PACKAGESTAGINGDIR\" \"${BUILDOUTPUTDIR}/$INSTALLERPACKAGENAME.pkg\""
+    echo $buildVFSForGitDistCmd
+    eval $buildVFSForGitDistCmd || exit 1
+    
+    /bin/rm -f "${BUILDOUTPUTDIR}/Distribution.updated.xml"
+}
```

Comment:
This method looks like it shares a lot of code from the `CreateMetaDistribution` method below.  Can we have a helper method that takes some parameters and shares most of this code?

Reply:
removed the duplication.